root/static/scripts/edit/components/URLInputPopover.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+import ButtonPopover from '../../common/components/ButtonPopover';
+
+type PropsT = {
+  errorMessage: string,
+  onCancel: () => void,
+  onChange: (number, SyntheticEvent<HTMLInputElement>) => void,
+  onToggle: (boolean) => void,
+  rawUrl: string,
+  url: string,
+};
+
+const URLInputPopover = (props: PropsT): React.MixedElement => {
+  const popoverButtonRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const toggle = (open) => {
+    /*
+     * Will be called by ButtonPopover when closed
+     * either by losing focus or click 'Close' button,
+     * therefore cancel action should be checked here.
+     */
+    if (!open) {
+      props.onCancel();
+    }
+    setIsOpen(open);
+    props.onToggle(open);
+  };
+
+  const onConfirm = () => {
+    // Bypass 'toggle' to avoid trigering onCancel
+    setIsOpen(false);
+    props.onToggle(false);
+  };
+
+  const buildPopoverChildren = (
+    closeAndReturnFocus,
+  ) => (
+    <form
+      onSubmit={(event) => {
+        event.preventDefault();
+        onConfirm();
+      }}
+    >
+      <table>
+        <tbody>
+          <tr>
+            <td className="section">
+              {addColonText(l('URL'))}
+            </td>
+            <td>
+              <input
+                className="value raw-url"
+                onChange={props.onChange}
```

Comment:
You're right, but the point here is to get instant error message while editing. The error is determined in ExternalLinkEditor and passed into ExternalLink, if we want to get error message directly in the popover, it would require a large refactoring.